### ISSUE TITLE
Exit command crashes iSH

### ISSUE THREAD

**Issue by Mnpn03** [ish-app/ish#816]
After looking at TestFlight feedback, I have found 17 reports that mention the `exit` command (and ctrl-d) crashing iSH with an EXC_BAD_ACCESS.
They all seem to have similar stack traces:
```
Thread 0 Crashed:
0   iSH                           	0x000000010212f81c tty_hangup + 28 (tty.c:771)
1   iSH                           	0x00000001020f6938 -[Terminal destroy] + 92 (Terminal.m:268)
2   iSH                           	0x00000001020eec10 -[TerminalViewController processExited:] + 192 (TerminalViewController.m:172)
3   CoreFoundation                	0x00000001bfd65644 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 20 (CFNotificationCenter.c:787)
4   CoreFoundation                	0x00000001bfd6568c ___CFXRegistrationPost1_block_invoke + 64 (CFNotificationCenter.c:175)
5   CoreFoundation                	0x00000001bfd649e4 _CFXRegistrationPost1 + 368 (CFNotificationCenter.c:198)
6   CoreFoundation                	0x00000001bfd646bc ___CFXNotificationPost_block_invoke + 88 (CFNotificationCenter.c:1371)
7   CoreFoundation                	0x00000001bfce1414 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1412 (CFXNotificationRegistrarOld.m:168)
8   CoreFoundation                	0x00000001bfd6401c _CFXNotificationPost + 1236 (CFNotificationCenter.c:1359)
9   Foundation                    	0x00000001c00c3974 -[NSNotificationCenter postNotificationName:object:userInfo:] + 60 (NSNotification.m:576)
10  iSH                           	0x00000001021046c0 __ios_handle_exit_block_invoke + 248 (AppDelegate.m:43)
11  libdispatch.dylib             	0x00000001bface9a8 _dispatch_call_block_and_release + 24 (init.c:1408)
12  libdispatch.dylib             	0x00000001bfacf524 _dispatch_client_callout + 16 (object.m:495)
13  libdispatch.dylib             	0x00000001bfab26fc _dispatch_main_queue_callback_4CF$VARIANT$armv81 + 860 (inline_internal.h:2484)
14  CoreFoundation                	0x00000001bfd877fc __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12 (CFRunLoop.c:1749)
15  CoreFoundation                	0x00000001bfd826d0 __CFRunLoopRun + 1724 (CFRunLoop.c:3069)
16  CoreFoundation                	0x00000001bfd81ce8 CFRunLoopRunSpecific + 424 (CFRunLoop.c:3192)
17  GraphicsServices              	0x00000001c9ecc38c GSEventRunModal + 160 (GSEvent.c:2246)
18  UIKitCore                     	0x00000001c3eb0444 UIApplicationMain + 1932 (UIApplication.m:4823)
19  iSH                           	0x00000001020fd840 main + 88 (main.m:13)
20  libdyld.dylib                 	0x00000001bfc098f0 start + 4
```

I could find reports dating as far back as build 65 in March, and I could replicate this on build 72.

I could only sometimes replicate this, and only on my phone - not in the simulator, and often only on my second try. The first time would just blank the screen and bring me back to a fresh prompt. Both iPhones and iPads have reported crashes, iOS 13 and 14 alike.

**Comment by thomasjo** [comment]
Race condition perhaps? Might explain why you're not able to reproduce it in the simulator.

**Comment by demhademha** [comment]
"Reboot" also causes ish to crash rather than restarting it.

**Comment by tbodt** [comment]
Fixed in 1eb7bc6ff16898a30dc7294ba8ffce3c52eead39
